packages/react-celo/src/utils/localStorage.ts

```diff
@@ -45,4 +50,93 @@ const localStorage =
     ? new MockedLocalStorage()
     : window.localStorage;
 
-export default localStorage;
+export function getLastUsedAddress(): string | null {
```

Comment:
If I understand correctly, you've done this to have better control over types and to serialize one of them?

One alternative to avoid having to create one function for each key is to have a generic getter.

If you transform `localStorageKeys` to an enum, you could do:
```ts
type KeyType<T> = T extends LocalStorageKeys.lastUsedFeeCurrency ? CeloTokenContract
  : T extends LocalStorageKeys.lastUsedWalletType ? WalletTypes
  : T extends LocalStorageKeys.lastUsedWalletArguments ? WalletArgs
  : string;

export function getTypedStorageKey<T extends LocalStorageKeys>(
  key: T
): KeyType<T> | null {
  const value = localStorage.getItem(key);
  if (!value) {
    return null;
  }

  if (key === LocalStorageKeys.lastUsedWalletArguments) {
    return JSON.parse(value) as KeyType<T>;
  }

  return localStorage.getItem(key) as KeyType<T>;
}

export function setTypedStorageKey<
  T extends LocalStorageKeys,
  V extends KeyType<T>
>(key: T, value: V): void {
  if (LocalStorageKeys.lastUsedWalletArguments) {
    const stringifiedValue = JSON.stringify(value);
    localStorage.setItem(key, stringifiedValue);
    return;
  }

  localStorage.setItem(key, value);
}
```

Or, you could have a simpler generic function for most keys:
```ts
export function getTypedStorageKey<T extends LocalStorageKeys>(
  key: T
): KeyType<T> | null {
  return localStorage.getItem(key) as KeyType<T>;
}

export function setTypedStorageKey<
  T extends LocalStorageKeys,
  V extends KeyType<T>
>(key: T, value: V): void {
  localStorage.setItem(key, value);
}
```
 and keep a specific one for `getLastUsedWalletArgs`.

Reply:
I also think it would be useful and straightforward to add tests for this, at the very least for the one that serializes.